src/lib.rs

```diff
@@ -1,32 +1,290 @@
+  /// Nodes must run the same DNA to be on the same DHT.
+  pub static ref APP_DNA_HASH: HashString = APP_GLOBALS.app_dna_hash.clone();
+
+  /// The identity string used to initialize this Holochain with `hcadmin init`.
+  /// If you used JSON to embed multiple properties (such as FirstName, LastName, Email, etc),
+  /// they can be retrieved here as App.Agent.FirstName, etc. (FIXME)
+  pub static ref APP_AGENT_ID_STR: String = APP_GLOBALS.app_agent_id_str.clone();
+
+  /// The hash of your public key.
+  /// This is your node address on the DHT.
+  /// It can be used for node-to-node messaging with `send` and `receive` functions.
+  pub static ref APP_AGENT_KEY_HASH: HashString = APP_GLOBALS.app_agent_key_hash.clone();
+
+  /// The hash of the first identity entry on your chain (The second entry on your chain).
+  /// This is your peer's identity on the DHT.
+  pub static ref APP_AGENT_INITIAL_HASH: HashString = APP_GLOBALS.app_agent_initial_hash.clone();
+
+  /// The hash of the most recent identity entry that has been committed to your chain.
+  /// Starts with the same value as APP_AGENT_INITIAL_HASH.
+  /// After a call to `update_agent` it will have the value of the hash of the newly committed identity entry.
+  pub static ref APP_AGENT_LATEST_HASH: HashString = APP_GLOBALS.app_agent_latest_hash.clone();
+}
+
+
+//--------------------------------------------------------------------------------------------------
+// SYSTEM CONSTS
+//--------------------------------------------------------------------------------------------------
+
+// HC.Version
+const VERSION : u16 = 1;
+const VERSION_STR : &'static str = "1";
+
+
+// HC.HashNotFound
+// FIXME keep in sync with HcApiReturnCode?
+pub enum ErrorCode {
+  FunctionNotImplemented,
+  HashNotFound,
+}
+
 
-type Hash = String;
+// HC.Status
+// WARNING keep in sync with CRUDStatus
+bitflags! {
+  pub struct EntryStatus: u8 {
+    const LIVE     = 1 << 0;
+    const REJECTED = 1 << 1;
+    const DELETED  = 1 << 2;
+    const MODIFIED = 1 << 3;
+  }
+}
 
-pub mod anchors;
 
-pub fn commit<S: Into<String>>(entry_type: S, entry: serde_json::Value) -> String {
-    "".to_string()
+// HC.GetMask
+bitflags! {
+  pub struct GetMask: u8 {
+    const ENTRY      = 1 << 0;
+    const ENTRY_TYPE = 1 << 1;
+    const SOURCES    = 1 << 2;
+  }
+}
+// explicit `Default` implementation
+impl Default for GetMask {
+    fn default() -> GetMask {
+        GetMask::ENTRY
+    }
 }
 
-pub fn link<S: Into<String>>(base: Hash, tag: S, target: Hash) {
 
+// HC.LinkAction
+pub enum LinkAction {
+    ADD,
+    DELETE,
 }
 
-pub fn get(entry_hash: Hash) -> Option<String> {
-    None
+
+// HC.PkgReq
+pub enum PkgRequest {
+    CHAIN,
+    CHAIN_OPTION,
+    ENTRY_TYPES,
 }
 
-pub fn get_links<S: Into<String>>(bash: Hash, tag: S) -> Vec<Hash> {
-    vec![]
+
+// HC.PkgReq.ChainOpt
+pub enum ChainOption {
+    NONE,
+    HEADERS,
+    ENTRIES,
+    FULL,
+}
+
+
+// HC.Bridge
+pub enum BridgeSide {
+    FROM,
```

Comment:
Is there still such a thing as bridging? I understood it was all moving to capabilities

Reply:
Yeah it will probably change in a near future. I'm focused on alpha2 and having the same api as the go version so I just copied it from the Go version.

Reply:
We already have API changes. Just sayin'.

Reply:
Not sure what you mean?